### PR TITLE
LibPDF+Meta: Use a CMYK ICC profile to convert CMYK to RGB

### PR DESCRIPTION
**Pull request by nico**
CMYK data describes which inks a printer should use to print a color.
If a screen should display a color that's supposed to look similar
to what the printer produces, it results in a color very different
to what Color::from_cmyk() produces. (It's also printer-dependent.)

There are many ICC profiles describing printing processes. It doesn't
matter too much which one we use -- most of them look somewhat
similar, and they all look dramatically better than Color::from_cmyk().

This patch adds a function to download a zip file that Adobe offers
on their web site. They even have a page for redistribution:
https://www.adobe.com/support/downloads/iccprofiles/icc_eula_win_dist.html

(That one leads to a broken download though, so this downloads the
end-user version.)

In case we have to move off this download at some point, there are also
a whole bunch of profiles at https://www.color.org/registry/index.xalter
that "may be used, embedded, exchanged, and shared without restriction".

The adobe zip contains a whole bunch of other useful and fun profiles,
so I went with it.

For now, this only unzips the USWebCoatedSWOP.icc file though, and
installs it in ${CMAKE_BINARY_DIR}/Root/res/icc/Adobe/CMYK/. In
Serenity builds, this will make it to /res/icc/Adobe/CMYK in the
disk image. And in lagom build, after https://github.com/SerenityOS/serenity/pull/23016 this is the
lagom res staging directory that tools can install via
Core::ResourceImplementation. `pdf` and `MacPDF` already do that,
`TestPDF` now does it too.

The final piece is that LibPDF then loads the profile from there
and uses it for DeviceCMYK color conversions.

(Doing file access from the bowels of a library is a bit weird,
especially in a system that has sandboxing built in. But LibGfx does
that in FontDatabase too already, and LibPDF uses that, so it's not a
new problem.)

---

The patch is probably smaller than that writeup :)

### CODE REVIEW

**Comment by nico** [comment]
Before:

Screenshot 2024-01-29 at 8 57 46 PM

After:

Screenshot 2024-01-29 at 4 20 40 PM

Before:

Screenshot 2024-01-29 at 8 57 39 PM

After:

Screenshot 2024-01-29 at 4 20 34 PM

**Comment by nico** [comment]
Another. Before:

Screenshot 2024-01-30 at 9 20 13 PM

After:

Screenshot 2024-01-30 at 9 19 55 PM

For reference, Preview.app:

0000206_p5_ref_preview

Chrome's PDF viewer:

0000206_p5_ref_pdfium

Firefox:

0000206_p5_ref_pdfjs

File: Meta/CMake/download_icc_profiles.cmake [hunk 1]
```diff
@@ -0,0 +1,21 @@
+include(${CMAKE_CURRENT_LIST_DIR}/utils.cmake)
+
+if (ENABLE_ADOBE_ICC_PROFILES_DOWNLOAD)
+    set(ADOBE_ICC_PROFILES_PATH "${SERENITY_CACHE_DIR}/AdobeICCProfiles" CACHE PATH "Download location for Adobe ICC profiles")
+    set(ADOBE_ICC_PROFILES_DATA_URL "https://download.adobe.com/pub/adobe/iccprofiles/win/AdobeICCProfilesCS4Win_end-user.zip")
+    set(ADOBE_ICC_PROFILES_ZIP_PATH "${ADOBE_ICC_PROFILES_PATH}/adobe-icc-profiles.zip")
+    download_file("${ADOBE_ICC_PROFILES_DATA_URL}" "${ADOBE_ICC_PROFILES_ZIP_PATH}")
```
Comment (ADKaster) [inline Meta/CMake/download_icc_profiles.cmake:7]: this call to download_file should be guarded by `if (ENABLE_NETWORK_DOWNLOADS)` like the other calls. It's my attempt to allow offline builds. Perhaps you have some improvements in that department as well :)

Comment (nico) [reply Meta/CMake/download_icc_profiles.cmake]: Done.

Do you have any particular improvements in mind?

(Maybe download_file() could contain the check for ENABLE_NETWORK_DOWNLOADS? Also, seems weird that the other files also guard the unzipping behind ENABLE_NETWORK_DOWNLOADS – not doing that for this one.)

**Comment by nico** [comment]
And one more, before:

Screenshot 2024-01-31 at 8 30 31 AM

After:

Screenshot 2024-01-31 at 8 32 02 AM

Compare how this used to look in https://github.com/SerenityOS/serenity/pull/21953#issuecomment-1813549317 ! (The other progression for this image making SMasks work for jpegs in #22026. That change was then made obsolete by #22650.)

**Review by ADKaster — APPROVED** [review]

**Pull request merged**